core/egov-mdms-service/src/main/java/org/egov/MDMSApplicationRunnerImpl.java

```diff
@@ -163,26 +160,32 @@ public void readMdmsConfigFiles(String masterConfigUrl) {
 		URL yamlFile = null;
 		try {
 			yamlFile = new URL(masterConfigUrl);
-		} catch (MalformedURLException e1) {
-			// TODO Auto-generated catch block
-			log.error("Exception while fetching service map for: " + e1.getMessage());
-		}
-		try {
 			file = jsonReader.readValue(new InputStreamReader(yamlFile.openStream()), Map.class);
-		} catch (Exception e) {
-			log.error("Exception while fetching service map for: " + e.getMessage());
+		} catch (IOException e1) {
+			log.error("Exception while fetching service map for: " + e1.getMessage());
 		}
-
 		masterConfigMap = file;
 		log.info("the Master config Map : " + masterConfigMap);
 
 	}
 
-	public static Map<String, Map<String, Map<String, JSONArray>>> getTenantMap() {
+	/***
+	 * 
+	 * method returns the tenantMap which has been loaded at the time of application
+	 * startup or refreshed during reload
+	 * 
+	 * @return tenantMap
+	 */
+	public  Map<String, Map<String, Map<String, JSONArray>>> getTenantMap() {
```

Comment:
Making this method public is somewhat dangerous.  By returning the full tenant map, you are basically allowing any outside class to mutate your internal state.  There are several alternative ways to do this:

1. Return a deep copy of the tenant map.  This almost certainly not what you want, since it can add significant performance overhead.
2. Make the tenant map unmodifiable before your return it, via the `Collections.unmodifiableMap` API.  This isn't a perfect solution, since the caller can still mutate the values in `tenantMap`, which are maps themselves.
3. Make this method package-local, so that at least it's not part of this class's public interface, but can still be called from other classes in the package.
4. Re-design the API so that you don't need to expose the tenant map to outside code at all.

For now, I would 3) above.  However, 4) would be the right solution down the road.